number_theory/utility.h

```diff
@@ -31,6 +32,38 @@ U binary_accumulate(T binary,
   return result;
 }
 
+// Returns the sign of the integer |x|.
+// If x = 0, sign(x) = 0.
+// If x > 0, sign(x) = 1.
+// If x < 0, sign(x) = -1.
+template <class T>
+constexpr int sign(T x) {
+  static_assert(std::numeric_limits<T>::is_integer,
+                "sign argument |x| must be an integer");
+
+  if (x == 0)
+    return 0;
+  return x > 0 ? 1 : -1;
+}
+
+// Returns the absolute value of the integer |x| in an unsigned type.
+//
+// This can be useful because std::abs doesn't support unsigned integers, and
+// std::abs(std::numeric_limits<T>::min()) is undefined.
+template <class T, class U = std::make_unsigned_t<T>>
+constexpr U unsigned_abs(T x) {
+  static_assert(std::numeric_limits<T>::is_integer,
+                "unsigned_abs argument |x| must be an integer");
+  static_assert(std::is_unsigned<U>::value,
```

Comment:
I suggest removing the return type from the template so that return type is always unsigned T.

Reply:
Also, we should prefer `std::numeric_limits::is_unsigned` over `std::is_unsigned` so that we can support user-defined types.